example/spec/ingestGranule/IngestGranuleDuplicateHandlingErrorCatchSpec.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const fs = require('fs-extra');
+const {
+  aws: { parseS3Uri },
+  stringUtils: { globalReplace }
+} = require('@cumulus/common');
+const {
+  buildWorkflow,
+  executeWorkflow,
+  LambdaStep
+} = require('@cumulus/integration-tests');
+const { api: apiTestUtils } = require('@cumulus/integration-tests');
+const {
+  loadConfig,
+  uploadTestDataToBucket,
+  deleteFolder,
+  timestampedTestDataPrefix
+} = require('../helpers/testUtils');
+const { setupTestGranuleForIngest } = require('../helpers/granuleUtils');
+
+const config = loadConfig();
+const lambdaStep = new LambdaStep();
+const workflowName = 'IngestGranuleCatchDuplicateErrorTest';
+
+const granuleRegex = '^MOD09GQ\\.A[\\d]{7}\\.[\\w]{6}\\.006\\.[\\d]{13}$';
+const testDataGranuleId = 'MOD09GQ.A2016358.h13v04.006.2016360104606';
+
+const s3data = [
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg'
+];
+
+describe('When the Ingest Granules workflow is configured to handle duplicates as "error"', () => {
+  const testDataFolder = timestampedTestDataPrefix(`${config.stackName}-IngestGranuleSuccess`);
+  const inputPayloadFilename = './spec/ingestGranule/IngestGranule.input.payload.json';
+  const collection = { name: 'MOD09GQ', version: '006' };
+  const provider = { id: 's3_provider' };
+  let workflowMessage;
+  let workflowExecution;
+  let inputPayload;
+  let granulesIngested;
+
+  beforeAll(async () => {
+    // upload test data
+    await uploadTestDataToBucket(config.bucket, s3data, testDataFolder, true);
+
+    const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+    // update test data filepaths
+    const updatedInputPayloadJson = globalReplace(inputPayloadJson, 'cumulus-test-data/pdrs', testDataFolder);
+    inputPayload = await setupTestGranuleForIngest(config.bucket, updatedInputPayloadJson, testDataGranuleId, granuleRegex);
+
+    workflowMessage = await buildWorkflow(
+      config.stackName, config.bucket, workflowName, collection, provider, inputPayload
+    );
+    // update duplicateHandling to 'error' for the collection in the workflow
+    workflowMessage.meta.collection.duplicateHandling = 'error';
```

Comment:
Ah, so this allows us to specify the duplicate handling for the execution without having to update an actual collection object?

Reply:
Yeah.  Because we are actually building workflow message here.  We won't be able to specify that in the case where workflow message is built using the collection configuration in s3/database, for example `queue-granules`